lib/cfg-grammar.y

```diff
@@ -1394,6 +1395,7 @@ vp_rekey_options
 
 vp_rekey_option
 	: KW_SHIFT '(' LL_NUMBER ')' { value_pairs_transform_set_add_func(last_vp_transset, value_pairs_new_transform_shift($3)); }
+	| KW_SHIFT_LEVELS '(' LL_NUMBER ')' { value_pairs_transform_set_add_func(last_vp_transset, value_pairs_new_transform_shift_levels($3)); }
```

Comment:
If I understand correctly the main use case for `shift-level` would be to cut the `--key` part, though it is capable more also it possible to misuse by accident. (The level of the key and the supplied value does not match.)

In that case I think it would be better to have an option like `--skip-key-prefix` which would automatically skip the key part.

Also the `shift-levels` has a side effect when the key name could be not unique anymore, which one should be preferred ? 
```
.a.b.c
.a.c.c
```
With shift level 3.

Reply:
Value-pairs would deduplicate the result, ie would use one of them, I think the last one.

But this is not necessarily a real problem. The convention in syslog-ng is to use an application specific prefix, so most value pairs would be unique under that prefix.

If there are exceptions (e.g. .unix or .SDATA) they can be handled separately.

Btw, the same can be true for --shift or any other key manipulation that removes parts from the key.

I don't understand --skip-key-prefix, can you elaborate on that?

Reply:
Based on this example: `$(format-json --key .SDATA.* --shift-levels 2)` and also on how I would use the `--shift-levels` option, for me it would be to drop the `.SDATA.` part of the key. If that is true, I think a simplified option could be used instead of `--shift-levels`.

Instead asking the user to count the dots in the key, syslog-ng could do the same with an option like `--skip-key-prefix` (maybe just `--skip-key`).

The `$(format-json --key <key> --skip-key)` would be equal to the `$(format-json --key <key> --shift-levels <number of dots in key>)`.

Reply:
Hmm.. nothing restricts syslog-ng name-value pairs to only have 1, 2 or 3 or more dots. Howlong prefix would --skip-key-prefix drop?

BTW: I tend to agree that value-pairs needs a serious reevaluation of its functionality, this is basically a quick hack to make name-value pairs map to Splunk nicely.

Reply:
It would use the string provided in the `--key` to determine the level, so in your example `--key .SDATA.*` it has 2 dots, therefore 2 level.

Example:
```
nv-pairs
a.b.c.d
a.b.d.e
a.c.g.f
```
The `--key a.b.* --skip-key` would result in: `c.d` and `d.e`.
The `--key a.b.c.* --skip-key` would result in `d`.

Reply:
ah.. that's actually a good idea. let me think about that.

Reply:
I gave it some thought, and could be implemented at this point as the wildcard would be available at the transformation rule.

My rationale:
1) I am generally unhappy with the value-pairs implementation and I am not sure the wildcard would be readily available to the transformation in the new one. This change does not introduce any new concepts/interdependencies in the command line syntax.

2) the wildcard might not be based on the prefix, in that case, it'd be difficult to decide what to do with --skip-key.

With that rationale, I'd keep the implementation as is. The value-pairs syntaxes are in a sorry state and should be overhauled at one point in the future. But since I am working something very different now, I wouldn't start that work. With this, I'd resolve this conversation.